File: February 2026 GFG SOLUTION/February-28.java
class Solution {
    public static ArrayList<Integer> findClosestPair(int arr1[], int arr2[], int x) {
        int i = 0, j = arr2.length - 1, minDiff = Integer.MAX_VALUE;
        ArrayList<Integer> res = new ArrayList<>();
        while (i < arr1.length && j >= 0) {
            int sum = arr1[i] + arr2[j], diff = Math.abs(sum - x);
            if (diff < minDiff) {
                minDiff = diff;
                res = new ArrayList<>(Arrays.asList(arr1[i], arr2[j]));
            }
            if (sum < x) i++;
            else j--;
        }
        return res;
    }
}

File: February 2026 GFG SOLUTION/February-28.py
class Solution:
    def findClosestPair(self, arr1, arr2, x):
        i, j, min_diff = 0, len(arr2) - 1, float('inf')
        res = []
        while i < len(arr1) and j >= 0:
            s, diff = arr1[i] + arr2[j], abs(arr1[i] + arr2[j] - x)
            if diff < min_diff:
                min_diff, res = diff, [arr1[i], arr2[j]]
            i, j = (i + 1, j) if s < x else (i, j - 1)
        return res
